modules/apps/organizations/organizations-service/src/main/java/com/liferay/organizations/internal/object/system/OrganizationSystemObjectDefinitionManager.java

```diff
@@ -0,0 +1,205 @@
+import com.liferay.portal.kernel.model.OrganizationTable;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.OrganizationLocalService;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Mateus Santana
+ */
+@Component(enabled = false, service = SystemObjectDefinitionManager.class)
+public class OrganizationSystemObjectDefinitionManager
+	extends BaseSystemObjectDefinitionManager {
+
+	@Override
+	public long addBaseModel(User user, Map<String, Object> values)
+		throws Exception {
+
+		OrganizationResource organizationResource = _buildOrganizationResource(
+			user);
+
+		Organization organization = organizationResource.postOrganization(
+			_toOrganization(values));
+
+		setExtendedProperties(
+			Organization.class.getName(), organization, user, values);
+
+		return GetterUtil.getLong(organization.getId());
+	}
+
+	@Override
+	public BaseModel<?> deleteBaseModel(BaseModel<?> baseModel)
+		throws PortalException {
+
+		return _organizationLocalService.deleteOrganization(
+			(com.liferay.portal.kernel.model.Organization)baseModel);
+	}
+
+	@Override
+	public BaseModel<?> fetchBaseModelByExternalReferenceCode(
+		String externalReferenceCode, long companyId) {
+
+		return _organizationLocalService.
+			fetchOrganizationByExternalReferenceCode(
+				externalReferenceCode, companyId);
+	}
+
+	@Override
+	public BaseModel<?> getBaseModelByExternalReferenceCode(
+			String externalReferenceCode, long companyId)
+		throws PortalException {
+
+		return _organizationLocalService.getOrganizationByExternalReferenceCode(
+			externalReferenceCode, companyId);
+	}
+
+	@Override
+	public String getExternalReferenceCode(long primaryKey)
+		throws PortalException {
+
+		com.liferay.portal.kernel.model.Organization organization =
+			_organizationLocalService.getOrganization(primaryKey);
+
+		return organization.getExternalReferenceCode();
+	}
+
+	@Override
+	public JaxRsApplicationDescriptor getJaxRsApplicationDescriptor() {
+		return new JaxRsApplicationDescriptor(
+			"Liferay.Headless.Admin.User", "headless-admin-user",
+			"organizations", "v1.0");
+	}
+
+	@Override
+	public Map<Locale, String> getLabelMap() {
+		return createLabelMap("organization");
+	}
+
+	@Override
+	public Class<?> getModelClass() {
+		return com.liferay.portal.kernel.model.Organization.class;
+	}
+
+	@Override
+	public List<ObjectField> getObjectFields() {
+		return Arrays.asList(
+			createObjectField(
+				"Text", "String", "comment", "comment", false, true),
+			createObjectField(
+				"Text", "String", "country-id", "countryId", false, true),
+			createObjectField("Text", "String", "name", "name", true, true),
+			createObjectField(
```

Comment:
Is it working? I'm not sure. How are we exposing `regionId` and `countryId` since we don't have it in the DTO schema?